catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceCallable.java

```diff
@@ -243,48 +243,60 @@ public ReliableResourceStatus call() {
 
                 bytesRead.addAndGet(n);
 
-                if (cacheFileOutputStream != null) {
-                    try {
-                        cacheFileOutputStream.write(buffer, 0, n);
-                    } catch (IOException e) {
-                        LOGGER.info("IOException during write to cached file's OutputStream", e);
-                        reliableResourceStatus =
-                                new ReliableResourceStatus(DownloadStatus.CACHED_FILE_OUTPUT_STREAM_EXCEPTION,
-                                        bytesRead.get());
-                    }
-                }
-
-                if (countingFbos != null) {
-                    try {
-                        countingFbos.write(buffer, 0, n);
-                        countingFbos.flush();
-                    } catch (IOException e) {
-                        LOGGER.info(
-                                "IOException during write to FileBackedOutputStream for client to read",
-                                e);
-                        reliableResourceStatus =
-                                new ReliableResourceStatus(DownloadStatus.CLIENT_OUTPUT_STREAM_EXCEPTION,
-                                        bytesRead.get());
-                    }
-                }
+                writeCacheFileOutputStream(buffer, n);
+                writeFileBackedOutputStream(buffer, n);
 
-                // Return status here so that each stream can be attempted to be updated regardless of
-                // which one might have had an exception
+                // Attempt to update each stream regardless of which one might have had an exception
                 if (reliableResourceStatus != null) {
                     return reliableResourceStatus;
                 }
             }
             LOGGER.trace("chunkCount = {},  bytesRead = {}", chunkCount, bytesRead.get());
         }
 
+        checkForSuccessfulCompletion();
+
+        return reliableResourceStatus;
+    }
+
+    private void writeCacheFileOutputStream(byte[] buffer, int n) {
+        if (cacheFileOutputStream != null) {
+            try {
+                cacheFileOutputStream.write(buffer, 0, n);
+            } catch (IOException e) {
+                LOGGER.warn("IOException during write to cached file's OutputStream", e);
+                updateReliableResourceStatus(DownloadStatus.CACHED_FILE_OUTPUT_STREAM_EXCEPTION);
+            }
+        }
+    }
+
+    private void writeFileBackedOutputStream(byte[] buffer, int n) {
+        if (countingFbos != null) {
+            try {
+                countingFbos.write(buffer, 0, n);
+                countingFbos.flush();
+            } catch (IOException e) {
+                LOGGER.info("IOException during write to FileBackedOutputStream for client to read",
```

Comment:
Is there a reason the `FileBackedOutputStream` exception is logged at info and cached file's `outputstream` exception is logged at warning?